### DISCUSSION TITLE
How to set different layout in different page?

### DISCUSSION THREAD

**Discussion by KahrLu** [nuxt/nuxt#17458] [Questions]
Hi @pi0 , due to we have several layout for several pages,  I want to control changing layout in app.vue.
```
<template>
  <NuxtLayout :name="layoutName">
    <NuxtPage />
  </NuxtLayout>
</template>

<script setup>
const route = useRoute()

const layoutName = computed(() => {
	return route.name === 'A' ? 'custom' : 'default'
})
</script>
```

then when I redirect from a page to A page, still cause the error below:
![image](https://user-images.githubusercontent.com/76502123/173032069-fe61de31-2788-4fcd-bde1-24f49938881d.png)

I'm sure it's not because I have multiple root nodes on **page A** (although it is one of the reasons for this error, now **page A** only have one root node)

When I remove computed, the error message disappears.

My requirement is that when switching from any page to **page A**, the layout is switched from default to custom
the example in the nuxt3 documentation also can't meet this requirement ， could you guys help me?
![image](https://user-images.githubusercontent.com/76502123/173032934-82de79c0-2214-4444-9ab4-62672852cc1e.png)

**Comment by Diizzayy** [comment]
@KahrLu can you provide a simple reproduction of what you're encountering? There might be something else within your code that's causing your error

The behavior that you're trying to implement is possible, and the layouts can indeed be changed solely via `app.vue`

You can see a live example here - https://n3l.vercel.app

`layouts/custom.vue`
```vue
<template>
  <div>
    <p style="color: red;">CUSTOM LAYOUT</p>
    <slot />
  </div>
</template>
```

`index.vue`
```vue
<template>
    <div>
        <h2>Home Page</h2>
    </div>
</template>
```

`special.vue`
```vue
<template>
    <h2>Special Page</h2>
</template>
```

`app.vue`
```vue
<template>
  <div>
    <nav>
      <ul>
        <li>
          <NuxtLink to="/">Home</NuxtLink>
        </li>
        <li>
          <NuxtLink to="/special">Special</NuxtLink>
        </li>
      </ul>
    </nav>
    <NuxtLayout :name="name">
      <NuxtPage />
    </NuxtLayout>
  </div>
</template>

<script lang="ts" setup>
const route = useRoute()

const name = computed(() => {
  if (route.name === 'special') {
    return 'custom'
  }

  return 'default'
})
</script>
```
